### ISSUE TITLE
PHP5 Type Hints

### ISSUE THREAD

**Issue by barchard** [spatie/schema-org#8]
Any thoughts on adding `TypeHints` to attributes that take another object instead of a `String`? E.g., https://github.com/spatie/schema-org/blob/master/src/FoodEstablishment.php#L69 which is of type  [Rating](https://github.com/spatie/schema-org/blob/master/src/Rating.php). I realize that there are some attributes which take a String or Object. Not sure the best approach is there.

**Comment by sebastiandedeyne** [comment]
I was considering this, decided it wasn't really worth it. My thoughts:

- It would add complexity to the generator, it would only allowed to add a type hint if there's only one possible property type
- Since most properties also allow strings, this would only apply to a small subset of properties
- It wouldn't provide any value when for the developer experience since the phpdocs already take care of that
- It's kind of a nice-to-have to be able to pass strings in—even though a type is required—if you're doing some custom stuff

In short: It wouldn't provide enough value imo. Also, adding this in now would be a breaking change in semver.
